defaults/main.yml

```diff
@@ -1,6 +1,7 @@
 ---
 elasticsearch_version: '7.x'
 elasticsearch_package_state: present
+elasticsearch_package_version: 7.10.2
```

Comment:
By specifying the package version explicitly, this means that every time someone wants to use this role, assuming I don't religiously keep this up to date with the latest version, they'd have to go find the latest version and override it.

I think the better option would just be to set `elasticsearch_package: elasticsearch`, and then people could override it with like `elasticsearch-7.10.2` if they want.

Reply:
Your suggestion is really great, but here is the limitations for that:

Current state of the role was using package module to install elasticsearch package [right here](https://github.com/geerlingguy/ansible-role-elasticsearch#L9). As i know, [package module](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/package_module.html) is a generic module and does not take any version arguments. So i had to change that module to [yum](https://github.com/bilalcaliskan/ansible-role-elasticsearch/blob/elastic-version/tasks/setup-RedHat.yml#L15) and [apt](https://github.com/bilalcaliskan/ansible-role-elasticsearch/blob/elastic-version/tasks/setup-Debian.yml#L20) modules to properly handle version arguments. I think we can not use a single variable called `elasticsearch_package: elasticsearch` because as you know, yum and apt modules are taking version information differently.

What about still defining a variable like `elasticsearch_package_version: latest` or with the any valid elasticsearch version as value in the [defaults/main.yml](https://github.com/bilalcaliskan/ansible-role-elasticsearch/blob/elastic-version/defaults/main.yml), and add 1 additional task to right [here](https://github.com/bilalcaliskan/ansible-role-elasticsearch/blob/elastic-version/tasks/setup-RedHat.yml#L15) and [here](https://github.com/bilalcaliskan/ansible-role-elasticsearch/blob/elastic-version/tasks/setup-Debian.yml#L20) which checks if the `elasticsearch_package_version` is `latest` or not. If not, installs a specific version by getting it from the variable, if it is `latest`, ignore the variable value which is latest and just install the elasticsearch package for both `os_family`.

Reply:
The point I'm making though, is that the `package` module just passes through the package name to `apt` or `yum` (or others), so if we just have the package name be a variable, Debian-style users could set `elasticsearch_package: "elasticsearch=x.y.z"`, while RedHat-style users could set `elasticsearch_package: "elasticsearch-x.y.z"`.

Reply:
first of all, i did not know that we can pass version information to the package module, so thank you for that. I have modified the code according to your review, can you review one more time?